src/npyffi/objects.rs

```diff
@@ -44,6 +57,139 @@ pub struct PyArray_Descr {
     pub hash: npy_hash_t,
 }
 
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct _PyArray_DescrNumPy2 {
```

Comment:
I would prefer a more consistent naming here, e.g. `_PyArray_Descr_NumPy1` and `_PyArray_Descr_NumPy2`. (`PyDataType_ISLEGACY` can stay as it is.)

Reply:
Done in #442.